wings/data_source/coinbase_pro_api_user_stream_data_source.py

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env python
+
+import asyncio
+import aiohttp
+import logging
+from typing import (
+    AsyncIterable,
+    Dict,
+    Optional,
+    List,
+)
+import ujson
+import websockets
+from websockets.exceptions import ConnectionClosed
+from wings.data_source.user_stream_tracker_data_source import UserStreamTrackerDataSource
+from wings.market.coinbase_pro_auth import CoinbaseProAuth
+from wings.order_book_message import OrderBookMessage
+from wings.orderbook.coinbase_pro_order_book import CoinbaseProOrderBook
+
+COINBASE_REST_URL = "https://api.pro.coinbase.com"
+COINBASE_WS_FEED = "wss://ws-feed.pro.coinbase.com"
+MAX_RETRIES = 20
+NaN = float("nan")
+
+
+class CoinbaseProAPIUserStreamDataSource(UserStreamTrackerDataSource):
+
+    MESSAGE_TIMEOUT = 30.0
+    PING_TIMEOUT = 10.0
+
+    _bausds_logger: Optional[logging.Logger] = None
```

Comment:
general convention seems to be to rename these based on file name. `_cbpausds_logger`

Reply:
sure